app/abci/handlers.go

```diff
@@ -88,10 +86,47 @@ func (h *Handlers) ExtendVoteHandler() sdk.ExtendVoteHandler {
 			}
 			return nil, err
 		}
+
+		if !h.signer.IsLoaded() {
+			h.logger.Info("signer is not loaded, try reloading")
+			err := h.signer.ReloadIfMismatch(nil)
+			if err != nil {
+				h.logger.Error("failed to load signer to sign batch", "err", err)
+				return nil, err
+			}
+		}
+
+		// Check if the validator was in the previous validator tree.
+		// If not, it means the validator just joined the active set,
+		// so it should start signing from the next batch.
+		_, err = h.batchingKeeper.GetValidatorTreeEntry(ctx, batch.BatchNumber-1, h.signer.GetValAddress())
+		if err != nil {
+			if errors.Is(err, collections.ErrNotFound) {
+				h.logger.Info("validator was not in the previous validator tree - not signing the batch")
+			} else {
+				h.logger.Error("unexpected error while checking previous validator tree entry", "err", err)
+			}
+			return nil, err
+		}
+
+		val, err := h.stakingKeeper.GetValidator(ctx, h.signer.GetValAddress())
+		if err != nil {
+			return nil, err
+		}
+		valKeys, err := h.pubKeyKeeper.GetValidatorKeys(ctx, val.OperatorAddress)
```

Comment:
Is the address we get from `h.signer.GetValAddress` not already the operator address? In my seda_keys.json I see `validator_addr: sedavaloper1ymuuw4cfa5j60hdecymfjkhsfhvxtt0ujfknvx`

Reply:
addressed in https://github.com/sedaprotocol/seda-chain/pull/413/commits/885432acd25f3dfbeaacb601da15dcdc8dbf5698